SimGeneral/Debugging/plugins/CaloParticleDebugger.cc

```diff
@@ -122,18 +127,48 @@ CaloParticleDebugger::analyze(const edm::Event& iEvent, const edm::EventSetup& i
 
   iEvent.getByToken(simTracksToken_, simTracksH);
   auto const & tracks = *simTracksH.product();
+  std::vector<int> sorted_tracks_idx(tracks.size());
+  iota(begin(sorted_tracks_idx), end(sorted_tracks_idx), 0);
+  sort(begin(sorted_tracks_idx),
+       end(sorted_tracks_idx),
+       [&tracks] (int i, int j) {
+        return tracks.at(i).momentum().eta() < tracks.at(j).momentum().eta();
```

Comment:
`[]` is slightly faster than `at()` and should be safe in this case (also for all below)